### ISSUE TITLE
Option to assume null for unset cells

### ISSUE THREAD

**Issue by sherman89** [nietras/Sep#164]
Hello @nietras and thanks for this library!

Would it be possible to have an option for `SepWriter` to assume that any non-written fields are null/empty?

I have a scenario in which the CSV data contains lists like this (simplified example)

![image](https://github.com/user-attachments/assets/0ad1d54f-28c2-4825-99a6-8b7852441f64)

In order to write CSV like this, I have to call `Set(null)` on the empty cells:
```csharp
for (int i = 0; i < rowCount; i++)
{
    using SepWriter.Row row = csvWriter.NewRow();

    if (i == 0)
    {
        row[0].Set(dto.Project);
        row[1].Set(dto.Author);
    }
    else
    {
        // Would like to get rid of this
        row[0].Set(null);
        row[1].Set(null);
    }

    if (i < contributorCount)
    {
        var contributor = dto.contributors[i];
        row[2].Set(contributor.Name);
    }
}
```

I could of course also just add this to the beginning of the new row:
```csharp
for (int c = 0; c < 2; c++)
{
    row[c].Set(null);
}
```

but the cleanest option is to let Sep handle this...no? 🙂

**Comment by nietras** [comment]
Hi, thanks for using Sep and suggesting this, however, this is the kind of thing that I think is counter to the philosophy of Sep and not something I would want to support directly. Sep is minimal with focus on being straightforward and efficient and not about reducing "your" code to minimal possible ;)

Not the least this code can be written straightforwardly and simply by:

```csharp
for (int i = 0; i < rowCount; i++)
{
    using var row = csvWriter.NewRow();
    row[0].Set(i == 0 ? dto.Project : string.Empty);
    row[1].Set(i == 0 ? dto.Author : string.Empty);
    row[2].Set(i < contributorCount ? dto.contributors[i].Name : string.Empty);
}
```
or similar e.g. if dto nullable then `dto?.Project ?? string.Empty`. 
Hope that is clear :)

**Comment by sherman89** [comment]
I understand, I just thought maybe it wouldn't be too complicated to not throw an exception if some cells were not set 😄

Your example is simpler but in my case it's more complicated and would result in very unreadable code.

In any case no worries, I've got it working with this simple loop:
```csharp
// Set null everywhere initially just to be safe from expected column errors
for (int colIndex = 0; colIndex < colCount; colIndex++)
{
    row[colIndex].Set(null);
}
```

**Comment by nietras** [comment]
@sherman89 I found a way to support the unset columns feature you have asked that fit within another issue, this is now available in 0.7.0.
